### PR TITLE
Run OPTIMADE validator for Netlify PR instance

### PR DESCRIPTION
**Pull request by CasperWA**
Use the [OPTIMADE validator GitHub Action](https://github.com/Materials-Consortia/optimade-validator-action) on the Netlify instance created for a PR to validate the providers as an index meta-database.

To do:
- [x] Test also all locally hosted index meta-databases.
- ~Possibly also test all externally hosted index meta-databases.~

### CODE REVIEW

**Comment by CasperWA** [comment]
@ml-evs I think the error occurring here is not the fault of the index meta-database, but rather our validator (see [here](https://github.com/Materials-Consortia/providers/pull/57/checks?check_run_id=1656306252#step:3:57)).

File: .github/workflows/validate.yml [hunk 1]
```diff
@@ -3,20 +3,33 @@ name: Scheduled validator
...
+  validator:
+    name: OPTIMADE validator
+    if: github.event_name == 'pull_request'
+    runs-on: ubuntu-latest
+    steps:
+    - uses: Materials-Consortia/optimade-validator-action@v2
+      with:
+        index: yes
+        protocol: https
+        domain: deploy-preview-${{ github.event.number }}--optimade-providers.netlify.app
```
Comment (ml-evs) [inline .github/workflows/validate.yml:33]: Can we try this as a temporary fix? (Of course, it should work until we release OPTIMADE version 2...)

```suggestion
        domain: deploy-preview-${{ github.event.number }}--optimade-providers.netlify.app/v1
```

Comment (CasperWA) [reply .github/workflows/validate.yml]: Sure. just pushed it, didn't notice your suggestion here, sorry.

**Comment by CasperWA** [comment]
Now it seems this PR is blocked by Materials-Consortia/optimade-python-tools#659.

**Comment by ml-evs** [comment]
The new error about missing provider prefixes should strongly encourage you to review and merge https://github.com/Materials-Consortia/optimade-python-tools/pull/659 :grin:

**Comment by ml-evs** [comment]
Right, so the new error is because netlify has a rich 404 page that doesn't have any JSON...

**Comment by ml-evs** [comment]
https://github.com/Materials-Consortia/optimade-python-tools/pull/665

**Comment by ml-evs** [comment]
Looking again at this, the validator itself runs correctly for the versioned URL without any changes; since the HTML 404 page still returns the expected 404 status code (duh), no attempt is made to deserialize to JSON in the validator, so the PR 665 linked above isn't super necessary (though it might be nice to have to give a more instructive error message when the 404 isn't expected). 

The problem here is that the validator *action* is taking the versioned URL as the _base URL_ and is appending *another* version to it. You can see in the logs that the first validation goes fine, but the second fails. I think this should be clarified in the validator action and will raise an issue there. For the meantime here, we can try the setting `all_versioned_paths: false`.

**Comment by CasperWA** [comment]
`all versioned paths` is by default set to `False`.
I don't think I'm completely following the issue, but if it's a problem of the validator action, please create an issue there. Never mind, just saw you created Materials-Consortia/optimade-validator-action#52 👍 Cheers.

**Comment by CasperWA** [comment]
The current reason for the validator failing is a _real_ reason. Issue #58 has been opened to track this.

**Comment by CasperWA** [comment]
I have now done stuff.
This now seems like a bit of being in a place between two worlds.
So what was called "Scheduled validator" wasn't set up to run on a schedule. I suspect this is because it's a copy-paste job from providers-dashboard.

Instead I have now separated the actual OPTIMADE validator tests into a separate workflow, which tests the Netlify deployed preview, both the list of providers index meta-db, as well as the locally hosted index meta-dbs.
This is done for all PRs.

Then the existing validation (pytest run) still runs for all PRs and all pushes. But I'm pretty sure the pytests are running similar tests as the OPTIMADE validator is running. So I'm suspecting we should simply remove it. But as I'm not 100 % sure, I'd rather do a 2-for-1 at this point.

**Review by ml-evs — APPROVED** [review]
LGTM, nice one @CasperWA.

I think its fine to keep the sanity checks in for now (mostly as a sanity check for the validator itself). It's a shame the wait-for-it hasn't been turned into an action yet!

**Comment by CasperWA** [comment]
> (...) It's a shame the wait-for-it hasn't been turned into an action yet!

Actually, I haven't even checked. There might be something. But yeah, otherwise it could be a procrastination project 😅

**Comment by ml-evs** [comment]
> > (...) It's a shame the wait-for-it hasn't been turned into an action yet!
> 
> Actually, I haven't even checked. There might be something. But yeah, otherwise it could be a procrastination project sweat_smile

I had a quick look but couldn't see anything, surprising given how widely that script is used

**Comment by CasperWA** [comment]
> > > (...) It's a shame the wait-for-it hasn't been turned into an action yet!
> > 
> > 
> > Actually, I haven't even checked. There might be something. But yeah, otherwise it could be a procrastination project sweat_smile
> 
> I had a quick look but couldn't see anything, surprising given how widely that script is used

Yeah! Okay. Well, I might set up something then. After hours :)

**Pull request merged**